### PR TITLE
Feature: add spot in the analysis results panel

### PR DESCRIPTION
**Pull request by raylrui**
## Feature

Add TSO500V2 tab in the analysis results panel.

## To Do
- [x]  check the apis for v2 content
- [x] refactor retrieve step for v2 step

### CODE REVIEW

**Comment by victorskl** [comment]
Ray; could you do me favour by create issue ticket & link to this, pls. You can use the same description. Tq

**Comment by raylrui** [comment]
Sure @victorskl, thanks for that.  Here #329  is the issue. I will add more details after.

**Comment by raylrui** [comment]
Accordingly, new tab will incloude `tsv`, `csv`, `vcf`, `json`, `bam` table to include all cttsov2 output files
image

**Review by reisingerf — APPROVED** [review]
LGTM

File: src/containers/subjects/AnalysisResultTable/index.tsx [hunk 1]
```diff
@@ -486,6 +496,23 @@ function groupResultsData({
     (r) => r.path.includes('tso_ctdna') && r.path.endsWith('tsv')
   );
 
+  // CTTSOV2 results
+  const cttsov2Bams = results_cttsov2.filter(
+    (r) => r.key.includes('cttsov2') && r.key.endsWith('bam')
+  );
+  const cttsov2Vcfs = results_cttsov2.filter(
+    (r) => r.key.includes('cttsov2') && (r.key.endsWith('vcf') || r.key.endsWith('vcf.gz'))
```
Comment (victorskl) [inline src/containers/subjects/AnalysisResultTable/index.tsx:504]: As discussed. Let filter `.vcf` to exclude `.gvcf` files.

Putting a dot should do it.
```
(r) => r.key.includes('cttsov2') && (r.key.endsWith('.vcf') || r.key.endsWith('.vcf.gz'))
```

File: src/containers/subjects/AnalysisResultTable/index.tsx
Comment (victorskl) [inline src/containers/subjects/AnalysisResultTable/index.tsx:1]: Could you also allow `vcf` to direct downloadable. It should be around Line 243.

```
const allowFileTypes = ['vcf', 'gz', 'maf', ...DATA_TYPE_SUPPORTED];
```

**Review by victorskl — APPROVED** [review]
Looking good. Minor comment to attend before merging.

Could you pls rebase to `dev`. There need regex tweak for `portal_run_id` extraction for BYOB key. It shows `cttsov2` now. :)

File: src/containers/subjects/AnalysisResultTable/index.tsx (hunk 1 above)
Comment (raylrui) [reply src/containers/subjects/AnalysisResultTable/index.tsx]: sure, added

File: src/containers/subjects/AnalysisResultTable/index.tsx
Comment (raylrui) [reply src/containers/subjects/AnalysisResultTable/index.tsx:1]: sure, added

**Review by williamputraintan — APPROVED** [review]

**Comment by raylrui** [comment]
Merging now

**Pull request merged**